### PR TITLE
Remove Mesos support

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:

Mesos had been introduced in the initial stages of Falco development. Afterward, it was never maintained and was rarely used by Falco users. As of today, we are not aware of any users or adopters using the Mesos support.

Behind the proposal of removing it, there're also other important aspects:
- there has not been any recent activity on the Mesos code (either in [Falco](https://github.com/falcosecurity/falco/search?q=mesos&type=code) and [libs](https://github.com/falcosecurity/libs/search?q=mesos&type=code))
- there have not been reported [issues](https://github.com/falcosecurity/falco/issues?q=is%3Aissue+is%3Aopen+mesos)
- there have not been discussions on Slack (I actually could not find any discussion)
- the support is not tested
- we are not sure if it's still fully working

**Special notes for your reviewer**:

/milestone 0.33.0

/hold
We want to gather some feedback on dropping the Mesos support before merging it.

**Does this PR introduce a user-facing change?**:

```release-note
update!: remove Mesos support
```

### CODE REVIEW

File: userspace/falco/app_actions/init_clients.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ using namespace falco::app;
 application::run_result application::init_clients()
 {
 #ifndef MINIMAL_BUILD
-	// k8s and mesos clients are useful only if syscall source is enabled
+	// k8s clients is useful only if syscall source is enabled
```
Comment (Andreagit97) [inline userspace/falco/app_actions/init_clients.cpp:24]:
```suggestion
	// k8s client is useful only if the syscall source is enabled
```

**Review by Andreagit97 — COMMENTED** [review]
LGTM!

File: userspace/falco/app_actions/init_clients.cpp (hunk 1 above)
Comment (leogr) [reply userspace/falco/app_actions/init_clients.cpp]: Good catch, thank you!

**Comment by 0snug0** [comment]
Currently we have several clients that have used and continue to use the mesos integration with a good amount of success. Given that mesos is not as fast of a release cycle or introducing as many breaking changes as Kubernetes, we have not seen many complaints, or had many feature request. 

I'd ask that we please leave the integration, as we can attest that is working

**Comment by leogr** [comment]
> Currently we have several clients that have used and continue to use the mesos integration with a good amount of success. Given that mesos is not as fast of a release cycle or introducing as many breaking changes as Kubernetes, we have not seen many complaints, or had many feature request.
> 
> I'd ask that we please leave the integration, as we can attest that is working

Hey @0snug0

Does your implementation depends directly on Falco or just on [libs](https://github.com/falcosecurity/libs)?
If the latter, we can unblock this and move the discussion to libs (since this PR only removes the support from Falco).

**Comment by leogr** [comment]
Closing and reopening to trigger the CI
/close

**Comment by leogr** [comment]
Closing in favor of https://github.com/falcosecurity/falco/pull/2328
/close

**Pull request closed without merging**